### DISCUSSION TITLE
can not add  multiple elements to an array in a if condition

### DISCUSSION THREAD

**Discussion by FogDong** [cue-lang/cue#1702] [Q&A] [answered]
### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
0.4.4
</pre>

### Does this issue reproduce with the latest release?

Yes

### What did you do?

```cue
a: "test"

b: [
  if a == "test" {
    "test1", "test2"
  }
]
```

```shell
$ cue eval test.cue
b.0: conflicting values "test2" and "test1":
    ./test.cue:5:5
    ./test.cue:5:14
```

### What did you expect to see?

```cue
b: ["test1", "test2"]
```

**Comment by myitcv** [comment, marked as answer]
@FogDong I converted your issue to a discussion because it's more of a question (although it might not appear to have been).

Please see https://github.com/cue-lang/cue/discussions/713

**Reply by FogDong** [reply]
Thanks for the quick answer. Though I think it's more like a workaround but it makes sense if it means `{ "test1" & "test2" }` in this case.

**Reply by myitcv** [reply]
Another slightly more fluent approach is to use `list.FlattenN()`:

```
import "list"

a: "test"

b: list.FlattenN([
	"test0",
	if a == "test" {
		["test1", "test2"]
	},
	if a == "test" {
		"test3"
	},
], 1)
```

[Playground link](https://cuelang.org/play/?id=UqEZ1hiehxW#cue@export@cue)

The [query extension](https://github.com/cue-lang/cue/issues/165) explores how this could be made a more native part of the language.

**Reply by dsaintilma-flinks** [reply]
@myitcv with this approach, any idea how could we also integrate a `...string` entry in case we want to add another list from another reference? I've tried on my end without any success.

**Reply by myitcv** [reply]
@dsaintilma-flinks I'm not entirely sure I follow your question. Please can you share an example?
